site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+.. _ByteArrayRequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/ByteArrayRequestConverterFunction.html
+.. _DecoratingServiceFunction: apidocs/index.html?com/linecorp/armeria/server/DecoratingServiceFunction.html
+.. _ExceptionHandlerFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/ExceptionHandlerFunction.html
+.. _HttpParameters: apidocs/index.html?com/linecorp/armeria/common/HttpParameters.html
+.. _HttpRequest: apidocs/index.html?com/linecorp/armeria/common/HttpRequest.html
+.. _HttpResponse: apidocs/index.html?com/linecorp/armeria/common/HttpResponse.html
+.. _HttpResponseException: apidocs/index.html?com/linecorp/armeria/server/HttpResponseException.html
+.. _HttpStatusException: apidocs/index.html?com/linecorp/armeria/server/HttpStatusException.html
+.. _JacksonRequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.html
+.. _PathMapping: apidocs/index.html?com/linecorp/armeria/server/PathMapping.html
+.. _Request: apidocs/index.html?com/linecorp/armeria/common/Request.html
+.. _RequestContext: apidocs/index.html?com/linecorp/armeria/common/RequestContext.html
+.. _RequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/RequestConverterFunction.html
+.. _ResponseConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/ResponseConverterFunction.html
+.. _ServerBuilder: apidocs/index.html?com/linecorp/armeria/server/ServerBuilder.html
+.. _Service: apidocs/index.html?com/linecorp/armeria/server/Service.html
+.. _ServiceRequestContext: apidocs/index.html?com/linecorp/armeria/server/ServiceRequestContext.html
+.. _StringRequestConverterFunction: apidocs/index.html?com/linecorp/armeria/server/annotation/StringRequestConverterFunction.html
+
+.. _server-annotated-service:
+
+Annotated HTTP Service
+======================
+
+Armeria provides a way to write an HTTP service using annotations. It helps a user make his or her code
+simple and easy to understand. A user is able to run an HTTP service by fewer lines of code using
+annotations as follows. ``getHello()`` method in the example would be mapped to the path of ``/hello``
+with an HTTP GET method.
+
+.. code-block:: java
+
+    ServerBuilder sb = new ServerBuilder();
+    sb.annotatedService(new Object() {
+        @Get("/hello/{name}")
+        public HttpResponse hello(@Param("name") String name) {
+            return HttpResponse.of(HttpStatus.OK,
+                                   MediaType.PLAIN_TEXT_UTF_8,
+                                   "Hello, %s!", name);
+        }
+    });
+
+Mapping HTTP service methods
+----------------------------
+
+To map a Java method in an annotated HTTP service class to HTTP path, it has to be annotated with one of
+HTTP method annotations. Here is the list of HTTP method annotations that Armeria supports. Each of them
```

Comment:
HTTP method annotations ... -> the following HTTP method annotations where each of them is ...: